### PR TITLE
New Interpreters and Setter, small improvement to Key setter

### PR DESCRIPTION
**Pull request by kubaplas**
New Interpreters:
- Asset - for adding assets existing in assets tree (in case of uploading them to Pimcore before import)
- DefaultObject - for setting existing object as a value of field
New Setter:
- AddToArray - for adding values to multi fields
Improvement in Key setter - using DataObject service' ValidKey method, instead of File's - now key can have spaces.

### CODE REVIEW

**Comment by aarongerig** [comment]
Thank you for your submission! We'll review your PR and will report back shortly.

File: src/ImportDefinitionsBundle/Interpreter/Asset.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+use ImportDefinitionsBundle\Model\Mapping;
+use Pimcore\Model\DataObject\Concrete;
+use ImportDefinitionsBundle\Service\Placeholder;
+
+/**
+ * Class Asset
+ *
+ * @package ImportDefinitionsBundle\Interpreter
+ */
+class Asset implements InterpreterInterface
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Interpreter/Asset.php]: I wouldn't name it Asset, cause it seems a bit to generic, asset could mean anything. I would rather call it AssetByPath

File: src/ImportDefinitionsBundle/Interpreter/Asset.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        Mapping $map,
+        $data,
+        DefinitionInterface $definition,
+        $params,
+        $configuration
+    ) {
+        $assetFullPath = $configuration['path'] . "/" . $value;
+        $asset = \Pimcore\Model\Asset::getByPath($assetFullPath);
+
+        if (!$asset) {
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Interpreter/Asset.php]: When Pimcore doesn't find an asset, it returns null anyway, so you could return the result from getByPath directly

File: src/ImportDefinitionsBundle/Interpreter/DefaultObject.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+use ImportDefinitionsBundle\Model\Mapping;
+use Pimcore\Model\DataObject;
+use Pimcore\Model\DataObject\Concrete;
+
+/**
+ * Class DefaultObject
+ *
+ * @package ImportDefinitionsBundle\Interpreter
+ */
+class DefaultObject implements InterpreterInterface
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Interpreter/DefaultObject.php]: I would'nt name it DefaultObject, as it expects to let you configure a "default" object. Something like ObjectByPath seems more right.

File: src/ImportDefinitionsBundle/Interpreter/DefaultObject.php [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    ) {
+        $path = $configuration['path'];
+
+        $object = DataObject::getByPath($path);
+
+        if ($object) {
+            return $object;
+        }
+
+        return false;
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Interpreter/DefaultObject.php]: Same as above for Asset, when pimcore doesn't find the object, it returns null anyway, so you can simply return the result from getByPath. Or is there any reason you return false here?

File: src/ImportDefinitionsBundle/Resources/config/services.yml [hunk 5]
```diff
@@ -186,5 +196,7 @@ services:
     tags:
       - { name: import_definition.setter, type: object_type, form-type: ImportDefinitionsBundle\Form\Type\NoConfigurationType }
 
-
-
+  import_definition.setter.add_to_array:
+    class: ImportDefinitionsBundle\Setter\ObjectType
```
Comment (dpfaffenbauer) [inline src/ImportDefinitionsBundle/Resources/config/services.yml]: Setter is not right configured here. Your class is named AddToArray and you configure ObjectType here.

Anyway: I don't see any sense why you are adding a new setter? Setter should actually apply the value the object/fieldcollection/localizedfield and so on. What you are doing is more like a InterpreterWrapper, right?

Comment (kubaplas) [reply src/ImportDefinitionsBundle/Resources/config/services.yml]: What's your proposition to handle objects and multihref fields?

Comment (dpfaffenbauer) [reply src/ImportDefinitionsBundle/Resources/config/services.yml]: chained interpreter would be a solution.

**Comment by dpfaffenbauer** [comment]
any updates on this?

**Comment by dpfaffenbauer** [comment]
@kubaplas could you please review the changes? if not, I'll close this PR

**Comment by dpfaffenbauer** [comment]
11 days, still no answers. Therefore I'll close this PR, if it is still necessary for you, please open it again and fix the issues.

**Pull request closed without merging**